### PR TITLE
register seats

### PR DESCRIPTION
**Pull request by namframgia**

### CODE REVIEW

**Comment by namframgia** [comment]
https://edu-redmine.framgia.vn/issues/4739

File: app/Http/Controllers/Admin/DiagramController.php [hunk 1]
```diff
@@ -64,15 +73,57 @@ public function saveWorkspace(Request $request)
...
-        $workspace = $this->workspace->findOrFail($idWorkspace);
-        $renderSeat = $this->renderSeat($workspace);
-        $locations = $workspace->locations;
-        $listProgram = $this->programRepository->listProgramArray();
-        $listPosition = $this->positionRepository->listpositionArray();
+        $user = Auth::user();
+        $firstDay = date('Y') . '-' . date('m') . '-01';
+        $lastDay = date('Y') . '-' . date('m') . '-' . date('t');
+        $workSchedules = WorkSchedule::where('user_id', $user->id)
+                            ->with(['seats' => function ($query) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/DiagramController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/DiagramController.php</br>**Line**: 82</br>Each line in an array declaration must end in a comma

File: app/Http/Controllers/Admin/DiagramController.php [hunk 2]
```diff
@@ -64,15 +73,57 @@ public function saveWorkspace(Request $request)
...
+                                $query->orderBy('schedule_seat.shift');
+                            }])
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/DiagramController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/DiagramController.php</br>**Line**: 84</br>Closing bracket of array declaration must be on a new line

File: app/Http/Controllers/SeatController.php [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        return $result;
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
```
Comment (framgiaciv3) [inline app/Http/Controllers/SeatController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/SeatController.php</br>**Line**: 111</br>String "-" does not require double quotes; use single quotes instead

File: app/Repositories/SeatRepository.php [hunk 4]
```diff
@@ -35,4 +38,77 @@ public function deleteSeat($id)
...
+    {
+        $saveData = [];
+        if (isset($requestData[$shift])) {
+            $data = $requestData[$shift];
+            foreach ($data as $key => $value) {
+                $saveData = array_merge($saveData, [[
+                    'work_schedule_id' => $key,
+                    'seat_id' => $value,
+                    'shift' => config('site.shift.' . $shift),
+                ]]);
```
Comment (framgiaciv3) [inline app/Repositories/SeatRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Repositories/SeatRepository.php</br>**Line**: 80</br>Comma required after last value in array declaration

Comment (framgiaciv3) [inline app/Repositories/SeatRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Repositories/SeatRepository.php</br>**Line**: 80</br>Closing bracket of array declaration must be on a new line

**Pull request merged**